lib/services/local/users.go

```diff
@@ -800,14 +807,53 @@ func (s *IdentityService) UpsertPassword(user string, password []byte) error {
 		return trace.Wrap(err)
 	}
 
-	err = s.UpsertPasswordHash(user, hash)
-	if err != nil {
+	if err := s.upsertPasswordHash(user, hash); err != nil {
 		return trace.Wrap(err)
 	}
 
+	_, err = s.UpdateAndSwapUser(
+		context.TODO(),
```

Comment:
```suggestion
		context.Background(),
```

I'd reserve TODO for placeholders that you intend to go back and address, and `Background` for places where you explicitly want a new context.